File: LaptopManagement/src/main/java/com/servlets/LoginServlet.java
package com.servlets;

import jakarta.servlet.ServletException;
import jakarta.servlet.annotation.WebServlet;
import jakarta.servlet.http.HttpServlet;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;

import java.io.IOException;
import java.io.PrintWriter;

/**
 * Servlet implementation class LoginServlet
 */
@WebServlet("/login")
public class LoginServlet extends HttpServlet {
	private static final long serialVersionUID = 1L;
       
    /**
     * @see HttpServlet#HttpServlet()
     */
    public LoginServlet() {
        super();
    }

	/**
	 * @see HttpServlet#doGet(HttpServletRequest request, HttpServletResponse response)
	 */
	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
		PrintWriter writer = response.getWriter();
		response.setCharacterEncoding("UTF-8");
		response.setContentType("text/html;charset=utf-8");
		
		writer.write("<html><body>");
		writer.write("<h1>登陆</h1>");
		writer.write("<form action='login' method='post'>");
		writer.write("<input type='text' name='username'> <input type='submit'>");
		if (request.getAttribute("error") != null && request.getAttribute("error").equals("usernameIsNull")) {
			writer.write("<font color='red'>用户名为空</font>");
		}
		writer.write("</form>");
		writer.write("</body></html>");
		writer.flush();
		writer.close();
	}

	/**
	 * @see HttpServlet#doPost(HttpServletRequest request, HttpServletResponse response)
	 */
	protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
		String username = request.getParameter("username").trim();
		if (username.equals("")) {
			request.setAttribute("error", "usernameIsNull");
			doGet(request, response);
		} else {
			request.getSession().setAttribute("username", username);
			response.sendRedirect("computer.html");
		}
	}

}

File: ChatGroupJsp/src/main/kotlin/com/lingfenglong/chatgroupjsp/servlets/LoginServlet.kt
package com.lingfenglong.chatgroupjsp.servlets

import jakarta.servlet.annotation.WebServlet
import jakarta.servlet.http.HttpServlet
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import java.util.HashSet

@WebServlet(name = "login", urlPatterns = ["/login"])
class LoginServlet : HttpServlet() {
    override fun doGet(req: HttpServletRequest?, resp: HttpServletResponse?) {
        super.doGet(req, resp)
    }

    override fun doPost(req: HttpServletRequest?, resp: HttpServletResponse?) {
        val username = req?.getParameter("username")
        val password = req?.getParameter("password")

        if (username != null && username.trim() != "") {
            val servletContext = this.servletContext
            var users = servletContext.getAttribute("users") as HashSet<String>?
            if (users == null) {
                users = HashSet<String>()
                servletContext.setAttribute("users", users)
            }
            users.add(username)

            req.session?.setAttribute("username", username)
            resp?.sendRedirect("main.html")
        } else {
            req?.setAttribute("errorInfo", "请填写您的账号信息")
            req?.getRequestDispatcher("index.jsp")?.forward(req, resp)
        }
    }
}